include/arch/arc/v2/arcv2_irq_unit.h

```diff
@@ -119,19 +119,13 @@ bool z_arc_v2_irq_unit_int_enabled(int irq)
  */
 
 static ALWAYS_INLINE
-void z_arc_v2_irq_unit_prio_set(int irq, unsigned char prio)
+void z_arc_v2_irq_unit_prio_set_raw(int irq, uint32_t prio)
 {
 
 	unsigned int key = arch_irq_lock();
 
 	z_arc_v2_aux_reg_write(_ARC_V2_IRQ_SELECT, irq);
-#if defined(CONFIG_ARC_SECURE_FIRMWARE)
```

Comment:
Why is this no longer needed? Is this _raw only used in case of ARC_CONNECT/SMP so Secure Mode is not relevant?

Reply:
Secure mode is still relevant, however this functionality has been split into two parts 
- the setting of raw value in hardware. It is done by `z_arc_v2_irq_unit_prio_set_raw`
- converting interrupt priority value into the raw value which should be written to HW (with taking care of secure bit and priority split for secure / normal). It is done by `ARC_IRQ_PRIO_TO_HW` macro.

PS: and yes, now this function is used only in interrupt controller init code.